keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/vetos/RequiredLoadBalancerVeto.kt

```diff
@@ -73,7 +74,13 @@ class RequiredLoadBalancerVeto(
     }
     spec.targetGroupsByRegion.forEach { (name, account, regions) ->
       val missingRegions = regions.filter { region ->
-        none { lb -> lb is ApplicationLoadBalancerModel && lb.account == account && lb.region == region && lb.targetGroups.any { it.targetGroupName == name } }
+        none { lb ->
+          when (lb) {
+            is ApplicationLoadBalancerModel -> lb.account == account && lb.region == region && lb.targetGroups.any { it.targetGroupName == name }
+            is NetworkLoadBalancerModel -> lb.account == account && lb.region == region && lb.targetGroups.any { it.targetGroupName == name }
+            else -> false
+          }
```

Comment:
Once again I wish Kotlin had intersection types

Reply:
if you were really bothered by the duplication here, you could derive an interface from `AmazonLoadBalancer` that has target groups on it (e.g., `TargetGroupedAmazonLoadBalancer`).